src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -381,12 +381,18 @@ public void setName(String name) {
      * @return interned string
      */
     protected String getCachedString(String str) {
-        WeakReference<String> internedString = stringCache.get(str);
-        if ( internedString == null || internedString.get() == null) {
-            internedString = new WeakReference<>(str);
-            stringCache.put(str, internedString);
+        if (!vcfIndexUseStringsCache)
+            return str;
+        WeakReference<String> stringWeakReference = stringCache.get(str);
```

Comment:
I totally agree with you @noamBarkai - we all have opinions, but a well-defined benchmark would be very handy.  To my way of thinking there are a couple of different things to benchmark:
1) A VCF with a a very high density of variants (maybe the exome VCF and one of the chromosome VCF for the genomes from http://gnomad.broadinstitute.org/downloads ?)
2) A VCF with a lot of genotyped samples - I'm not really sure what's out there for this other than some 1KG VCFs

Then the question is what is the benchmark on these files?  I think I'd suggest two benchmarks:
1) Reading/decoding the entire file and doing something very simple, e.g. counting the total number of genotypes 
2) Read the file and re-writing it - perhaps even sorting it?

If you're up for doing some of these I'm happy to work with you @yfarjoun to try and specify some benchmarks more clearly.